back/src/forms/resolvers.ts

```diff
@@ -34,6 +68,68 @@ export default {
       const dbForm = await context.prisma.form({ id });
       return unflattenObjectFromDb(dbForm);
     },
+
+    formLifeCycle: async (
+      _,
+      { siret, createdAfter, createdBefore, cursorAfter, cursorBefore, formId },
+      context: GraphQLContext
+    ) => {
+      const userId = context.user.id;
+      const userCompanies = await getUserCompanies(userId);
+      // User must be associated with a company
+      if (!userCompanies.length) {
+        throw new Error(
+          "Vous n'êtes pas autorisé à consulter le cycle de vie des bordereaux."
+        );
+      }
+      // If user is associated with several companies, siret is mandatory
+      if (userCompanies.length > 1 && !siret) {
+        throw new Error(
+          "Vous devez préciser pour quel siret vous souhaitez consulter"
+        );
+      }
+      // Select user company matching siret or get the first
+      const selectedCompany =
+        userCompanies.find(uc => uc.siret === siret) || userCompanies.shift();
+
+      const formsFilter = {
+        OR: [
+          { owner: { id: userId } },
+          { recipientCompanySiret: selectedCompany.siret },
+          { emitterCompanySiret: selectedCompany.siret },
+          {
+            transporterCompanySiret: selectedCompany.siret,
+            status: "SEALED"
+          }
+        ]
+      };
+
+      const statusLogsCx = await context.prisma
+        .statusLogsConnection({
+          orderBy: "created_DESC",
+          first: PAGINATE_BY,
+          after: cursorAfter,
```

Comment:
Ce serait bien de rajouter un index sur le champ `created`. Pas sûr que ce soit gérer pas prisma il faudra peut-être le faire à la mano

Reply:
Que se passe-t-il si on ajoute à la fois `first` et `after` ?

Reply:
J'aurais tendance à penser que `first` fonctionne en duo avec `after` et que `last` fonctionne en duo avec `before`. En effet la phrase suivante ne fait pas vraiment sens: "Give me the last 100 items after X".

Reply:
> Ce serait bien de rajouter un index sur le champ `created`. Pas sûr que ce soit géré par prisma il faudra peut-être le faire à la mano
Si ça se révèle nécessaire, faudra ajouter l'index à la main
> Que se passe-t-il si on ajoute à la fois first et after

Tu obtiens les x(first) elts après l'id référencé par after 
cf  https://prisma-docs.netlify.com/docs/1.0/reference/prisma-api/queries-ahwee4zaey/#pagination

> J'aurais tendance à penser que first fonctionne en duo avec after et que last fonctionne en duo avec before. En effet la phrase suivante ne fait pas vraiment sens: "Give me the last 100 items after X".
Je suis pas sûr de comprendre, y'a pas de  paramètre last ici